zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js

```diff
@@ -87,40 +108,56 @@ export const getProfile = (refs, props) => {
  */
 export const editProfile = (e, props, toast) => {
   e.preventDefault();
+  let password_match = true;
   if (props.values.user_location.length < 1) {
     props.validateField('user_location');
+  } else if (props.values.password.length < 1) {
+    props.validateField('editProfile.inputs.password.errors');
   } else {
-    return props
-      .editUserProfile({ ...props.values, token: props.auth.token })
-      .then(_ => {
-        toast.success(props.t('editProfile.toastSuccess'));
-        props.history.push('/profile');
-      })
-      .catch(error => {
-        const messages = JSON.parse(error.message);
-        if (typeof messages === 'object') {
-          const server_errors = {};
-          Object.keys(messages).forEach(key => {
-            if (key === 'non_field_errors') {
-              server_errors['non_field_errors'] = messages[key][0];
-            } else if (key === 'location') {
-              server_errors['user_location'] = messages[key][0];
+    props.login({ values: props.values, history: props.history }).catch(error => {
+      const messages = JSON.parse(error.message);
+      console.log(messages);
```

Comment:
remove console.log

Reply:
And there might be parsing error if error.message is not JSON. try to surround the block with try catch

Reply:
FYI you can use async and awaits to fix the callbacks and doing password_match checks

Reply:
@chaitanya-cloudknox Updated the PR for the first 2 comments. I actually tried using async and await only initially but it was always giving me undefined saying promise unresolved so had to use .finally and implement this way.